diff --git a/lib/functions/configuration/main-config.sh b/lib/functions/configuration/main-config.sh
index 887f7d729..658c4778f 100644
--- a/lib/functions/configuration/main-config.sh
+++ b/lib/functions/configuration/main-config.sh
@@ -8,6 +8,8 @@
 # https://github.com/armbian/build/
 
 ### Attention: we can't use any interactive programs, read from stdin, nor use non-coreutils utilities here.
+REGIONAL_MIRROR=china
+GHCR_MIRROR=dockerproxy
 
 function do_main_configuration() {
 	display_alert "Starting main configuration" "${MOUNT_UUID}" "info"
@@ -151,8 +153,8 @@ function do_main_configuration() {
 	case $REGIONAL_MIRROR in
 		china)
 			[[ -z $USE_MAINLINE_GOOGLE_MIRROR ]] && [[ -z $MAINLINE_MIRROR ]] && MAINLINE_MIRROR=tuna
-			[[ -z $USE_GITHUB_UBOOT_MIRROR ]] && [[ -z $UBOOT_MIRROR ]] && UBOOT_MIRROR=gitee
-			[[ -z $GITHUB_MIRROR ]] && GITHUB_MIRROR=gitclone
+			[[ -z $USE_GITHUB_UBOOT_MIRROR ]] && [[ -z $UBOOT_MIRROR ]] && UBOOT_MIRROR=github
+			[[ -z $GITHUB_MIRROR ]] && GITHUB_MIRROR=ghproxy
 			[[ -z $DOWNLOAD_MIRROR ]] && DOWNLOAD_MIRROR=china
 			;;
 		*) ;;
@@ -224,7 +226,7 @@ function do_main_configuration() {
 			GHCR_SOURCE='ghcr.dockerproxy.com'
 			;;
 		*)
-			GHCR_SOURCE='ghcr.io'
+			GHCR_SOURCE='ghcr.dockerproxy.com'
 			;;
 	esac
 
diff --git a/lib/functions/general/oci-oras.sh b/lib/functions/general/oci-oras.sh
index cc19fdf54..9d507af88 100644
--- a/lib/functions/general/oci-oras.sh
+++ b/lib/functions/general/oci-oras.sh
@@ -55,7 +55,7 @@ function run_tool_oras() {
 
 	declare ORAS_FN="oras_${ORAS_VERSION}_${ORAS_OS}_${ORAS_ARCH}"
 	declare ORAS_FN_TARXZ="${ORAS_FN}.tar.gz"
-	declare DOWN_URL="https://github.com/oras-project/oras/releases/download/v${ORAS_VERSION}/${ORAS_FN_TARXZ}"
+	declare DOWN_URL="${GITHUB_SOURCE}/oras-project/oras/releases/download/v${ORAS_VERSION}/${ORAS_FN_TARXZ}"
 	declare ORAS_BIN="${DIR_ORAS}/${ORAS_FN}"
 	declare ACTUAL_VERSION
 
diff --git a/lib/functions/general/shellcheck.sh b/lib/functions/general/shellcheck.sh
index 2cb46659f..012b07503 100644
--- a/lib/functions/general/shellcheck.sh
+++ b/lib/functions/general/shellcheck.sh
@@ -94,7 +94,7 @@ function run_tool_shellcheck() {
 
 	declare SHELLCHECK_FN="shellcheck-v${SHELLCHECK_VERSION}.${SHELLCHECK_OS}.${SHELLCHECK_ARCH}"
 	declare SHELLCHECK_FN_TARXZ="${SHELLCHECK_FN}.tar.xz"
-	declare DOWN_URL="https://github.com/koalaman/shellcheck/releases/download/v${SHELLCHECK_VERSION}/${SHELLCHECK_FN_TARXZ}"
+	declare DOWN_URL="${GITHUB_SOURCE}/koalaman/shellcheck/releases/download/v${SHELLCHECK_VERSION}/${SHELLCHECK_FN_TARXZ}"
 	declare SHELLCHECK_BIN="${DIR_SHELLCHECK}/${SHELLCHECK_FN}"
 	declare ACTUAL_VERSION
 
